api/v1alpha1/virtualmachine_types.go

```diff
@@ -392,6 +392,41 @@ type NetworkInterfaceStatus struct {
 	IpAddresses []string `json:"ipAddresses,omitempty"`
 }
 
+// NetworkInterfaceIP defines the network interface IP configuration including gateway, subnetmask and IP address
+// as seen by OVF properties
+type NetworkInterfaceIP struct {
```

Comment:
This type is _off_. Every `NetworkDevice` should have one or more IP addresses and a gateway. The subnet is not needed as the IP addresses should have a CIDR length attached to them.